app/views/cloud_object_store_container/new.html.haml

```diff
@@ -0,0 +1,62 @@
+%form#form_div{:name => "angularForm", 'ng-controller' => "cloudObjectStoreContainerFormController"}
+  = render :partial => "layouts/flash_msg"
+  %h3
+    = _('Basic Information')
+  .form-horizontal
+    .form-group{"ng-class" => "{'has-error': angularForm.storage_manager_id.$invalid}"}
+      %label.col-md-2.control-label
+        = _('Storage Manager')
+      .col-md-8
+        = select_tag("storage_manager_id",
+          options_for_select([["<#{_('Choose')}>", nil]] + @storage_manager_choices.sort, disabled: ["<#{_('Choose')}>", nil]),
+                      "ng-model"                    => "cloudContainerModel.storage_manager_id",
+                      "ng-change"                   => "storageManagerChanged(cloudContainerModel.storage_manager_id)",
+                      "required"                    => "",
+                      :miqrequired                  => true,
+                      :checkchange                  => true,
+                      "selectpicker-for-select-tag" => "")
+
+    .form-group
+      %label.col-md-2.control-label
+        = _('Container Name')
+      .col-md-8
+        %input.form-control{:type          => "text",
+                            :name          => "name",
+                            'ng-model'     => "cloudContainerModel.name",
+                            'ng-maxlength' => 255,
+                            :miqrequired   => true,
+                            :checkchange   => true}
+  %div{"ng-if" => "cloudContainerModel.emstype == 'ManageIQ::Providers::Amazon::StorageManager::S3'"}
+    %h3
+      = _('Placement')
+    .form-horizontal
+      .form-group{"ng-class"  => "{'has-error': angularForm.provider_region.$invalid}",
+                  "ng-if"     => "#{@provider_regions.keys.to_json}.includes(cloudContainerModel.parent_emstype)",
+                  "ng-switch" => "",
+                  "on"        => "cloudContainerModel.parent_emstype"}
+        %label.col-md-2.control-label{"for" => "ems_region"}
+          = _('Region')
+        .col-md-8
+          - @provider_regions.each do |name, regions|
+            = select_tag('provider_region',
```

Comment:
Same goes here... an API would be preferable.
Will get back to you on the exact API that you can use later.